spec/acceptance/trusted_external_data_spec.rb

```diff
@@ -119,4 +119,27 @@
       expect(trusted_json['external']['servicenow']['fqdn']).to eql(master.uri)
     end
   end
+
+  # This test provides coverage for oauth tokens both encrypted and not.
+  context 'user specifies a hiera-eyaml encrypted oauth token' do
+    # skip the oauth tests if we don't have a token to test with
+    servicenow_config = servicenow_instance.bolt_config['remote']
```

Comment:
Yeah, for this one we want something like this:
```
skip_oauth_tests = false
unless <using_mock_instance>
  skip_oauth_tests = (servicenow_config['oauth_token']) ? false : true
end
```

The reason is b/c we are skipping oauth token tests on real SNow instance b/c setting one up can be a PITA. But on the mock instance you don't need to set it up at all so there's no good reason for skipping the tests there.

I think simple test for `<using_mock_instance>` is `using_mock_instance = servicenow_instance.uri =~ Regexp.new(Regexp.escape(master.uri))`

Reply:
Also, were you able to get oauth token stuff working on mock instance? If I recall, the mock instance currently accepts user/password authentication only so you may need to update https://github.com/puppetlabs/puppetlabs-servicenow_cmdb_integration/blob/master/spec/support/acceptance/servicenow/mock_instance.rb#L23-L26.

Note mock instance is (currently) necessary for CI until we can get a persistent dev instance.